Models/reader.rb

```diff
@@ -1,11 +1,34 @@
+# frozen_string_literal: true
+
 class Reader
-  attr_accessor name, email, city, street, house
+  attr_reader :name, :email, :city, :street, :house
+
+  def initialize(args)
+    validate(args)
+    @name = args[:name]
+    @email = args[:email]
+    @city = args[:city]
+    @street = args[:street]
+    @house = args[:house]
+  end
+
+  def to_s
+    "#{name} - #{email}. Address: #{city}, #{house} #{street}"
+  end
+
+  private
+
+  def validate_str(name)
+    raise StandardError, "#{name} is incorrect" unless name.is_a?(String) && name != ''
```

Comment:
If `name` will be blank, then you receive `" is incorrect"` error message

Reply:
def validate_str(name = 'Text parameter') was addded